### DISCUSSION TITLE
fkYAML version 0.3.10

### DISCUSSION THREAD

**Discussion by fktn-k** [fktn-k/fkYAML#373] [Announcements]
## Summary

This release adds the new support for deserializing/serializing multiple YAML documents by adding new APIs ([`fkyaml::node::deserialize_docs()`](https://fktn-k.github.io/fkYAML/api/basic_node/deserialize_docs/) and [`fkyaml::node::serialize_docs()`](https://fktn-k.github.io/fkYAML/api/basic_node/serialize_docs/) respectively). You can still call the existing APIs for deserializing/serializing a single YAML document. See the linked API document pages for details.  
In addition, from this version on, Intel icpx compiler is supported and used in the CI processes. Some compiler flags are added in building the unit testing app, but no compiler specific swiches are required for the library itself.  
Moreover, because the GitHub Actions runner image for macOS11 has been deprecated, the CI jobs which uses the runner image and some compiler support which depend on the runner image have been removed.  
For other changes like bug fixes, see descriptions in each related issues and PRs.  

### :sparkles: New Features

- Support Intel icpx compiler by @fktn-k in #360
- Support parssing multiple YAML documents by @fktn-k in #362
- Support serializing multiple YAML docs by @fktn-k in #363

### :bug: Bug Fixes

- Detect missing the end of directives markers (---) by @fktn-k in #361
- Fix bug in serializing alias keys by @fktn-k in #364
- Allow trailing comma in flow mapping/sequence by @fktn-k in #365
- \#368 Fix line advancement after node props by @fktn-k in #369, reported by @alienczf in #368
- Apply node properties for the root node by @fktn-k in #370
- \#366 Fix node-to-float conversion error if node's value is \<= 0 by @fktn-k in #371, reported by @ARessegetesStery in #366

### :robot: CI

- Remove CI jobs running with macOS 11 by @fktn-k in #367
- update configs for reuse v4 by @fktn-k in #372

**Full Changelog**: https://github.com/fktn-k/fkYAML/compare/v0.3.9...v0.3.10

<hr /><em>This discussion was created from the release <a href='https://github.com/fktn-k/fkYAML/releases/tag/v0.3.10'>fkYAML version 0.3.10</a>.</em>

**Comment by joelmandell** [comment]
**Copilot analysis:** Great idea! A user guide based on our Blazor components would be very helpful. The codebase-structure.json already maps 312 components. I could generate a markdown guide with TOC automatically.
